CONTRIBUTUNG.md

```diff
@@ -0,0 +1,44 @@
+Please always report problems by filling an issue here on GitHub. This also covers reporting problems with the documentation (e.g. if sth is unclear).
+
+General rules:
+
+    1. the code style suggestions of PyCharm should be followed
+
+Job general:
+
+    1. If a class inherits directly/indirectly from the sisyphus class `Job` the class ends with "Job".
+    2. In a Sisyphus job class the output variables end in `out_*`
+    3. If a Job has non-trivial requirements it should have a `self.rqmt`
+
+Job constructor order:
+
+    1. set inputs
+    2. set outputs (should be prefixed with `out_`)
+    3. define rqmt
+
+Job function order:
+
+    1. `__init__`
+    2. `tasks()`
+    3. task functions
+    4. helper functions
+    5. class functions
+    6. `__hash__`
+
+Rasr-Jobs:
+should always accept `crp`, `extra_config` + `extra_post_config`
+
+Import order:
+
+    1. standard library
+    2. external libraries (including sisyphus, yes I know it's not like that atm)
+    3. recipies
+
+Some general things:
+
+    1. Don't break a jobs hash
+    2. Really don't break a jobs hash
+    3. very strong preference to avoid config files / scripts in the recipe repo itself. Everything should be code
+    4. Jobs should be as deterministic as possible
```

Comment:
I would formulate this as "jobs have to be deterministic, if this is not possible at all add a clear warning"

Reply:
Almost no jobs are 100% deterministic in practice. I'm sure when you run many things involving `float` at some point on different hardware, you will get different results.

And obviously, all our trainings are very much non-deterministic.

Reply:
What I want to say is, this is a case which will very often happen (every training job), and we should also be prepared and be able to deal with this.

Reply:
Float operations are also deterministic most of the time, or at least deterministic enough to not cause issues. Only in NN training we do so many of them to lead to different results.

Reply:
I still would choose a stricter formulation. Of course hardware non-determinism can not be fixed in any way, but at least algorithmic non-determinism (random operations need a seed, no order of lines should change, number of output etc...) should not occur in any case.

Reply:
I remember that we got different results for GMM splitting depending on amd64 vs x86 although we used float32 in both cases.